lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -226,29 +226,31 @@ export default class DocumentComposer {
   }
 
   private static validateRemovePublicKeysPatch (patch: any) {
-    const patchProperties = Object.keys(patch);
-    if (patchProperties.length !== 2) {
-      throw new SidetreeError(ErrorCode.DocumentComposerPatchMissingOrUnknownProperty);
+    const allowedProperties = new Set(['action', 'ids']);
+    for (const property in patch) {
+      if (!allowedProperties.has(property)) {
+        throw new SidetreeError(ErrorCode.DocumentComposerUnknownPropertyInRemovePublicKeysPatch, `Unexpected property ${property} in remove-public-keys patch.`);
+      }
     }
 
-    if (!Array.isArray(patch.publicKeys)) {
+    if (!Array.isArray(patch.ids)) {
       throw new SidetreeError(ErrorCode.DocumentComposerPatchPublicKeyIdsNotArray);
     }
 
-    for (const publicKeyId of patch.publicKeys) {
-      if (typeof publicKeyId !== 'string') {
-        throw new SidetreeError(ErrorCode.DocumentComposerPatchPublicKeyIdNotString);
-      }
+    for (const id of patch.ids) {
+      DocumentComposer.validateId(id);
     }
   }
 
   /**
    * validate update patch for removing services
    */
   private static validateRemoveServicesPatch (patch: any) {
-    const patchProperties = Object.keys(patch);
-    if (patchProperties.length !== 2) {
-      throw new SidetreeError(ErrorCode.DocumentComposerPatchMissingOrUnknownProperty);
+    const allowedProperties = new Set(['action', 'ids']);
+    for (const property in patch) {
+      if (!allowedProperties.has(property)) {
+        throw new SidetreeError(ErrorCode.DocumentComposerUnknownPropertyInRemoveServicesPatch, `Unexpected property ${property} in remove-services patch.`);
+      }
     }
 
     if (!Array.isArray(patch.ids)) {
```

Comment:
Do we also need a check to see if action exists? Since we removed the length check, we no longer knows if a patch has both, we just know it doesn't contain extra keys.

Reply:
I think it's okay, because these private method are invoked by `validatePatch` which already does that check.